src/EditorFeatures/Core/Wrapping/SeparatedSyntaxList/SeparatedSyntaxListCodeActionComputer.cs

```diff
@@ -0,0 +1,376 @@
+                //     int d, int e, int f, int g,
+                //     int h, int i, int j)
+                codeActions.Add(await GetWrapLongLineCodeActionAsync(
+                    parentTitle, WrappingStyle.UnwrapFirst_IndentRest).ConfigureAwait(false));
+
+                // The wrap-all and wrap-long code action titles are not unique.  i.e. we show them
+                // as:
+                //      Wrap every parameter:
+                //          Align parameters
+                //          Indent wrapped parameters
+                //      Wrap long parameter list:
+                //          Align parameters
+                //          Indent wrapped parameters
+                //
+                // We can't in-line these nested actions because the parent title is necessary to
+                // determine which situation each child action applies to.
+
+                return new WrappingGroup(isInlinable: false, codeActions.ToImmutableAndFree());
+            }
+
+            private async Task<WrapItemsAction> GetWrapLongLineCodeActionAsync(
+                string parentTitle, WrappingStyle wrappingStyle)
+            {
+                var indentationTrivia = GetIndentationTrivia(wrappingStyle);
+
+                var edits = GetWrapLongLinesEdits(wrappingStyle, indentationTrivia);
+                var title = GetNestedCodeActionTitle(wrappingStyle);
+
+                return await TryCreateCodeActionAsync(edits, parentTitle, title).ConfigureAwait(false);
+            }
+
+            private ImmutableArray<Edit> GetWrapLongLinesEdits(
+                WrappingStyle wrappingStyle, SyntaxTrivia indentationTrivia)
+            {
+                var result = ArrayBuilder<Edit>.GetInstance();
+
+                AddTextChangeBetweenOpenAndFirstItem(wrappingStyle, result);
+
+                var currentOffset = wrappingStyle == WrappingStyle.WrapFirst_IndentRest
+                    ? indentationTrivia.FullWidth()
+                    : _afterOpenTokenIndentationTrivia.FullWidth();
+                var itemsAndSeparators = _listItems.GetWithSeparators();
+
+                for (var i = 0; i < itemsAndSeparators.Count; i += 2)
+                {
+                    var item = itemsAndSeparators[i].AsNode();
+
+                    // Figure out where we'd be after this item.
+                    currentOffset += item.Span.Length;
+
+                    if (i > 0)
+                    {
+                        if (currentOffset < WrappingColumn)
+                        {
+                            // this item would not make us go pass our preferred wrapping column. So
+                            // keep it on this line, making sure there's a space between the previous
+                            // comma and us.
+                            result.Add(Edit.UpdateBetween(itemsAndSeparators[i - 1], SingleWhitespaceTrivia, NoTrivia, item));
+                            currentOffset += " ".Length;
```

Comment:
? why not just += 1?

Reply:
This is self-documenting.  '1' means nothing.  But `" ".Length` makes it clear it's the length of that space.